From: Heiko Stuebner <heiko@sntech.de>
Subject: [PATCH] debian/fix-eagain-gcc

Fix useless warning
/tmp/hdctools-20160523/lib/ftdiuart.c: In function ‘fuart_wr_rd_locked’:
/tmp/hdctools-20160523/lib/ftdiuart.c:362:45: error: logical ‘or’ of equal expressions [-Werror=logical-op]
     if ((bytes == -1) && ((errno == EAGAIN) || (errno == EWOULDBLOCK)) &&
                                             ^~
cc1: all warnings being treated as errors


See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=69602 for reference

Signed-off-by: Heiko Stuebner <heiko@sntech.de>

---
 lib/ftdiuart.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ftdiuart.c b/lib/ftdiuart.c
index 7d1054c..c461462 100644
--- a/lib/ftdiuart.c
+++ b/lib/ftdiuart.c
@@ -359,7 +359,7 @@ static int fuart_wr_rd_locked(struct fuart_context *fuartc) {
       rd_buf += bytes;
       bytes_remaining -= bytes;
     }
-    if ((bytes == -1) && ((errno == EAGAIN) || (errno == EWOULDBLOCK)) &&
+    if ((bytes == -1) && (errno == EAGAIN || (EWOULDBLOCK != EAGAIN && errno == EWOULDBLOCK)) &&
         (retries < 10)) {
       retries++;
       goto retry_write;
-- 
tg: (f2558e6..) debian/fix-eagain-gcc (depends on: upstream)
